src/librustc_passes/ast_validation.rs

```diff
@@ -167,11 +168,58 @@ impl<'a> AstValidator<'a> {
                 "only lifetime parameters can be used in this context");
         }
     }
+
+    /// With eRFC 2497, we need to check whether an expression is ambigious and warn or error
+    /// depending on the edition, this function handles that.
+    fn while_if_let_ambiguity(&self, expr: &P<Expr>) {
+        if let Some((span, op_kind)) = self.while_if_let_expr_ambiguity(&expr) {
+            let mut err = self.err_handler().struct_span_err(
+                span, &format!("ambigious use of `{}`", op_kind.to_string())
+            );
+
+            err.note(
+                "this will be a error until the `let_chains` feature is stabilized"
+            );
```

Comment:
Perhaps we should talk about direct people to https://github.com/rust-lang/rust/issues/53668 to leave a comment?

Something like:

```rust
err.note(
    "see rust-lang/rust#53668 for more information"
);
```

Reply:
Yes, that sounds good (including the exact snippet you wrote). 👍

Reply:
Added this.